libnmstate/nm/bridge.py

```diff
@@ -165,36 +176,57 @@ def _get_bridge_setting(nmdev):
     return bridge_setting
 
 
-def _get_bridge_ports_info(port_profiles, vlan_filtering_enabled=False):
-    ports_info = []
-    for p in port_profiles:
-        port_info = _get_bridge_port_info(p)
+def _get_bridge_ports_info(
+    port_names_profiles, port_runtime_names, vlan_filtering_enabled=False
+):
+    ports_name_info = {}
+    for name in port_runtime_names:
+        port_info = _get_bridge_port_runtime_config(name)
+        ports_name_info[name] = port_info
+
+    for name, p in port_names_profiles:
+        port_info = ports_name_info.get(name, {})
         if port_info:
             if vlan_filtering_enabled:
                 bridge_vlan_config = p.get_setting_bridge_port().props.vlans
                 port_vlan = PortVlanFilter()
                 port_vlan.import_from_bridge_settings(bridge_vlan_config)
                 port_info[LB.Port.VLAN_SUBTREE] = port_vlan.to_dict()
-            ports_info.append(port_info)
-    return ports_info
-
-
-def _get_bridge_port_info(port_profile):
-    """Report port information."""
-
-    port_setting = port_profile.get_setting_bridge_port()
-    return {
-        LB.Port.NAME: port_profile.get_interface_name(),
-        LB.Port.STP_PRIORITY: port_setting.props.priority,
-        LB.Port.STP_HAIRPIN_MODE: port_setting.props.hairpin_mode,
-        LB.Port.STP_PATH_COST: port_setting.props.path_cost,
-    }
+    return [port_info for port_info in ports_name_info.values()]
 
 
-def _get_slave_profiles(master_device):
-    slave_profiles = []
+def _get_slave_names_profiles(master_device):
+    slaves_names_profiles = []
     for dev in master_device.get_slaves():
         active_con = connection.get_device_active_connection(dev)
         if active_con:
-            slave_profiles.append(active_con.props.connection)
-    return slave_profiles
+            slave_name_profile = (dev.get_iface(), active_con.props.connection)
+            slaves_names_profiles.append(slave_name_profile)
```

Comment:
This is ugly IMO. Can you make this a dict (i.e. `slaves_profiles_by_name`) ?

Reply:
Yes, done thanks!